### ISSUE TITLE
Local value in the template is overriding style setter

### ISSUE THREAD

**Issue by wieslawsoltes** [AvaloniaUI/Avalonia#2098]
The style is getting higher priority but local value in the template is overriding style properties:
```XAML
        <Style Selector="NumericUpDown.picker /template/ TextBox#PART_TextBox">
            <Setter Property="Margin" Value="4"/>
        </Style>
```
```XAML
<NumericUpDown Classes="picker"/>
```
https://github.com/AvaloniaUI/Avalonia/blob/7dea76262b180e26b59918a09ce89667d9921a26/src/Avalonia.Themes.Default/NumericUpDown.xaml#L32

![bez tytulu](https://user-images.githubusercontent.com/2297442/48478841-c016cb80-e805-11e8-91da-efbac7228daa.png)

**Comment by Gillibald** [comment]
Is Margin="{TemplateBinding Margin}" with a setter in the style working?

**Comment by grokys** [comment]
Yes, this is a known problem and in fact the same behavior occurs in WPF (and UWP as far as I know).

Problem is that if you're setting a property using `Property="Value"` you're setting the property with the `LocalValue` priority, exactly as if you were setting the property in a `Window` etc. This means that a style won't be able to override it as the style will be setting the property with the `Style` or `StyleTrigger` priority, which is lower than `LocalValue`.

I'd really like to find a way to make properties set in templates use the `Style` priority instead of `LocalValue`, but I can't quite think of a way to achieve this!

**Comment by maxkatz6** [comment]
@grokys in UWP you can override values from template in the VisualStateGroup.
But actually VisualStateGroup should be defined in same template so it isn't perfect solution, but it is possible.
In Avalonia there is no any in-template VisualStateGroup equivalent, but only second style with specific selector, which doesn't work in that case.

**Comment by grokys** [comment]
Fix by #7679.
